stl/src/syserror_import_lib.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#include <__msvc_system_error_abi.hpp>
+
+#include <Windows.h>
+
+namespace {
+
+    struct _Whitespace_bitmap_t {
+        bool _Is_whitespace[256];
+
+        constexpr _Whitespace_bitmap_t() : _Is_whitespace{} {
+            _Is_whitespace[' ']  = true;
+            _Is_whitespace['\n'] = true;
+            _Is_whitespace['\r'] = true;
+            _Is_whitespace['\t'] = true;
+            _Is_whitespace['\0'] = true;
+        }
+
+        constexpr bool _Test(const char _Ch) const noexcept {
+            return _Is_whitespace[static_cast<unsigned char>(_Ch)];
+        }
+    };
+
+    constexpr _Whitespace_bitmap_t _Whitespace_bitmap;
+
+} // unnamed namespace
+
+_EXTERN_C
+
+_NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_get_string_size_without_trailing_whitespace(
+    _In_ const char* const _Str, _In_ size_t _Size) noexcept {
+    if (_Size != 0) {
+        do {
+            --_Size;
+        } while (_Whitespace_bitmap._Test(_Str[_Size]));

```

Comment:
I feel like this could buffer underflow on an all whitespace string.

Reply:
Hmm I assumed that was impossible but I agree. How about:

    for (;;) {
        if (_Size == 0) {
            return _Size;
        }

        --_Size;
        if (!_Whitespace_bitmap._Test(_Str[_Size])) {
            ++_Size;
            return _Size;
        }
    }

Reply:
I've had this somewhat more compact version:
```cpp
while (--_Size != 0) {
    if (!_Whitespace_bitmap._Test(_Str[_Size])) {
        return ++_Size;
    }
}

return _Size;
```

Reply:
That still needs a guard since _Size == 0 on entry will go bad :)

Reply:
Good point! I'll just use yours (this function benefit from unit testing but I guess it's a write-once kind of thing)

Reply:
It's in the header so a test would be a good idea. I'll add some.